pkg/i2gw/providers/gce/gce.go

```diff
@@ -32,16 +35,26 @@ func init() {
 
 // Provider implements the i2gw.Provider interface.
 type Provider struct {
-	storage   *storage
-	reader    reader
-	converter converter
+	storage      *storage
+	reader       reader
+	converter    converter
+	extConverter extensionConverter
 }
 
 func NewProvider(conf *i2gw.ProviderConf) i2gw.Provider {
+	if conf.Client != nil {
```

Comment:
I believe this check is in place to differentiate for when we are reading from a file, mind adding a short comment?

Reply:
Sounds good! Added.